### PR TITLE
fix(Android): Report correct port, even when automatically allocated

### LINKED ISSUE #81: TcpServer doesn't report correct port when it is automatically allocated
## Description

According to the Java documentation of ServerSocket ( https://docs.oracle.com/javase/7/docs/api/java/net/ServerSocket.html#ServerSocket(int,%20int,%20java.net.InetAddress) ) 
if you provide 0 for the port number a port will be automatically allocated. That part works correctly in react-native-tcp-socket, but the 'connect' callback reports the port to be 0, when in fact that is not the actual port it's listening on. It should get the port number after the ServerSocket instance is created by calling `getLocalPort`.

This works correctly on iOS.

## Current behavior
Port is reported as 0 when the port passed to tcpServer.listen is 0.

## Expected behavior
The actual port number should be returned.

## Relevant information
|             |     |
| ----------- | --- |
| OS |  Android  |
| react-native | 0.62   |
| react-native-tcp-socket  | ?   |

### PR DESCRIPTION
**Pull request by eddieAnthem**
Fixes #81

### CODE REVIEW

File: android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java [hunk 1]
```diff
@@ -103,4 +107,8 @@ public void close() {
             mReceiverListener.onClose(getId(), e.getMessage());
         }
     }
+
+    public int getListeningPort() {
+        return listeningPort;
```
Comment (Rapsssito) [inline android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java:112]: There is no need for a `listeningPort` attribute. `serverSocket.getLocalPort()` returns the port number to which this socket is listening or -1 if the socket is not bound yet. You could just use `serverSocket.getLocalPort()` here.

File: android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java [hunk 2]
```diff
@@ -103,4 +107,8 @@ public void close() {
...
+    }
```
Comment (Rapsssito) [inline android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java:113]: There is no need for a `listeningPort` attribute. `serverSocket.getLocalPort()` returns the port number to which this socket is listening or -1 if the socket is not bound yet. You could just use `serverSocket.getLocalPort()` here.

**Review by Rapsssito — CHANGES_REQUESTED** [review]

**Comment by Rapsssito** [comment]
@eddieAnthem, thanks for your PR! I have reviewed your code and requested some changes, but it looks good to me!

File: android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java (hunk 2 above)
Comment (eddieAnthem) [reply android/src/main/java/com/asterinet/react/tcpsocket/TcpSocketServer.java]: Good point. I have updated it as you suggested.

**Comment by Rapsssito** [comment]
LGTM!

**Pull request merged**